include/boost/gil/image_processing/morphology.hpp

```diff
@@ -0,0 +1,285 @@
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+#define BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/gray.hpp>
+#include <boost/gil/image_processing/threshold.hpp>
+
+namespace boost {
+namespace gil {
+namespace detail {
+  enum class morphological_operation {
+  dilation,
+  erosion,
+};
+/// \addtogroup ImageProcessing
+/// @{
+
+/// \brief Implements morphological operations at pixel level.This function
+/// compares neighbouring pixel values according to the kernel and choose
+/// minimum/mamximum neighbouring pixel value and assigns it to the pixel under
+/// consideration. \param src_view - Source/Input image view. \param dst_view -
+/// View which stores the final result of operations performed by this function.
+/// \param kernel - Kernel matrix/structuring element containing 0's and 1's
+/// which will be used for applying the required morphological operation. \param
+/// identifier - Indicates the type of morphological operation to be applied.
+/// \tparam SrcView type of source image.
+/// \tparam DstView type of output image.
+/// \tparam Kernel type of structuring element.
+template <typename SrcView, typename DstView, typename Kernel>
+void morph_impl(SrcView const &src_view, DstView const &dst_view,
+                Kernel const &kernel, morphological_operation identifier) {
+  std::ptrdiff_t flip_ker_row, flip_ker_col, row_boundary, col_boundary;
+  typename channel_type<typename SrcView::value_type>::type target_element;
+  for (std::ptrdiff_t view_row = 0; view_row < src_view.height(); ++view_row) {
+    for (std::ptrdiff_t view_col = 0; view_col < src_view.width(); ++view_col) {
+      target_element = src_view(view_col, view_row);
+      for (std::size_t kernel_row = 0; kernel_row < kernel.size();
+           ++kernel_row) {
+        flip_ker_row =
+            kernel.size() - 1 - kernel_row; // row index of flipped kernel
+
+        for (std::size_t kernel_col = 0; kernel_col < kernel.size();
+             ++kernel_col) {
+          flip_ker_col =
+              kernel.size() - 1 - kernel_col; // column index of flipped kernel
+
+          // index of input signal, used for checking boundary
+          row_boundary = view_row + (kernel.center_y() - flip_ker_row);
+          col_boundary = view_col + (kernel.center_x() - flip_ker_col);
+
+          // ignore input samples which are out of bound
+          if (row_boundary >= 0 && row_boundary < src_view.height() &&
+              col_boundary >= 0 && col_boundary < src_view.width()) {
+            // We ensure that we consider only those pixels which are overlapped
+            // on a non-zero kernel_element as
+            if(kernel.at(flip_ker_row,flip_ker_col) == 0){
+              continue;
```

Comment:
When mentioning this improvement I meant placing it on line 53 from current commit.

Reply:
Okay,I will change it